Bottom: 33cef57ba150958a3d5c55c3149f722f984cae4d
Top:    ebbbea4baf1eb2d90c2ec262de6265ed8b91b556
Author: Jason Wang <jasowang@redhat.com>
Date:   2018-05-29 17:21:53 +0800

Refresh of vhost_net-split-out-datacopy

---

diff --git a/drivers/vhost/net.c b/drivers/vhost/net.c
index 787cd72..ed04efa 100644
--- a/drivers/vhost/net.c
+++ b/drivers/vhost/net.c
@@ -537,11 +537,10 @@ static void handle_tx_copy(struct vhost_net *net)
 
 		total_len += len;
 		if (total_len < VHOST_NET_WEIGHT &&
-		    vhost_has_more_pkts(net, vq)) {
+		    !vhost_vq_avail_empty(&net->dev, vq))
 			msg.msg_flags |= MSG_MORE;
-		} else {
+		else
 			msg.msg_flags &= ~MSG_MORE;
-		}
 
 		/* TODO: Check specific error and bomb out unless ENOBUFS? */
 		err = sock->ops->sendmsg(sock, &msg, len);
